src/webots/nodes/WbMesh.cpp

```diff
@@ -304,8 +308,9 @@ void WbMesh::updateTriangleMesh(bool issueWarnings) {
 }
 
 uint64_t WbMesh::computeHash() const {
-  const QByteArray meshPathNameIndex = (path() + mName->value() + mMaterialIndex->value()).toUtf8();
-  return WbTriangleMeshCache::sipHash13x(meshPathNameIndex.constData(), meshPathNameIndex.size());
+  QString meshPathNameIndex = path();
+  meshPathNameIndex += (mIsCollada ? mName->value() + QString::number(mMaterialIndex->value()) : "");
+  return WbTriangleMeshCache::sipHash13x(meshPathNameIndex.toUtf8().constData(), meshPathNameIndex.size());
```

Comment:
Can we do this instead?

```suggestion
  const QByteArray meshHash = (path() + (mIsCollada ? mName->value() + mMaterialIndex->value()) : "").toUtf8();
  return WbTriangleMeshCache::sipHash13x(meshHash.constData(), meshHash.size());
```

Reply:
Another suggestion above is better that mine.